.github/workflows/self-assign.yml

```
@@ -34,10 +35,15 @@ jobs:
           curl -H "Authorization: token $GITHUB_TOKEN" -d '{"assignees":["'"$LOGIN"'"]}' https://api.github.com/repos/$REPO/issues/$ISSUE_NUMBER/assignees
           curl -X DELETE -H "Authorization: token $GITHUB_TOKEN" https://api.github.com/repos/$REPO/issues/$ISSUE_NUMBER/labels/awaiting%20triage
         fi
-        if [[ $BODY == *"$INPUT_CLOSE"* && $BODY != *"\`$INPUT_CLOSE\`"* ]]; then
-          echo "Closing issue $ISSUE_NUMBER"
+        ISSUE_ASSIGNEE="$(jq '.issue.assignee.login' $GITHUB_EVENT_PATH | tr -d \")"
+        if [[ $ISSUE_ASSIGNEE == $LOGIN && $BODY == *"$INPUT_CLOSE"* && $BODY != *"\`$INPUT_CLOSE\`"* ]]; then
```

```suggestion
        if [[ $BODY == *"$INPUT_CLOSE"* && $BODY != *"\`$INPUT_CLOSE\`"* ]]; then
```

I'm sorry, I misspoke (mistyped?) earlier, though my reasoning was still correct. I don't think we should restrict issue close to the assignee. Closing an issue as a non-assignee is probably fairly common (e.g. an owner of a migrated ticket needs to close it), and its not doing anything permissions-wise since anyone can self-assign. I can't think of a scenario where restricting like this helps us.

---

Agree. adjusted